server/src/main/java/org/candlepin/controller/ContentManager.java

```diff
@@ -75,45 +116,56 @@ public ContentManager(ContentCurator contentCurator, ProductCurator productCurat
      *  given owner
      *
      * @throws NullPointerException
-     *  if the provided content entity is null
+     *  if entity or owner is null
      *
      * @return
      *  a new Content instance representing the specified content for the given owner
      */
     @Transactional
     public Content createContent(Content entity, Owner owner) {
-        log.debug("Creating new content for org: {}, {}", entity, owner);
-
         if (entity == null) {
-            throw new NullPointerException("entity");
+            throw new IllegalArgumentException("entity is null");
         }
 
-        Content existing = this.contentCurator.lookupById(owner, entity.getId());
+        if (owner == null) {
+            throw new IllegalArgumentException("owner is null");
+        }
+
+        log.debug("Creating new content for org: {}, {}", entity, owner);
+
+        Content existing = this.ownerContentCurator.getContentById(owner, entity.getId());
+
+        // TODO: FIXME:
```

Comment:
This TODO and the matching one in ProductManager are referring to the this bug that's triggered if the create method is called with an entity that already exists. I suppose a quick way to address the TODO is to mark the createContent(entity, owner) method as protected so it can't be called directly.